src/main/scala/com/github/mrpowers/spark/fast/tests/DataframeUtil.scala

```diff
@@ -0,0 +1,115 @@
+package com.github.mrpowers.spark.fast.tests
+
+import com.github.mrpowers.spark.fast.tests.ufansi.Color.{DarkGray, Green, Red}
+import com.github.mrpowers.spark.fast.tests.ufansi.Str
+import org.apache.commons.lang3.StringUtils
+import org.apache.spark.sql.Row
+
+object DataframeUtil {
+
+  def showDataframeDiff(
+      header: (String, String),
+      actual: Array[Row],
+      expected: Array[Row],
+      truncate: Int = 20
+  ): String = {
+
+    val sb = new StringBuilder
+
+    val a = actual.zipAll(expected, Row(), Row())
+    val diff = a.map { case (a, e) =>
+      if (equals(a, e)) {
+        List(ufansi.Color.DarkGray(a.toString()), ufansi.Color.DarkGray(e.toString()))
+      } else {
+        val actual   = a.toSeq
+        val expected = e.toSeq
+        if (actual.isEmpty)
+          List(
+            Red("[]"),
+            Green(expected.mkString("[", ",", "]"))
+          )
+        else if (expected.isEmpty)
+          List(Red(actual.mkString("[", ",", "]")), Green("[]"))
+        else {
+          val withEquals = actual
+            .zip(expected)
+            .map { case (a1, e1) => (a1, e1, a1 == e1) }
+          val allFieldsAreNotEqual = !withEquals.exists(_._3)
+          if (allFieldsAreNotEqual) {
+            List(
+              Red(actual.mkString("[", ",", "]")),
+              Green(expected.mkString("[", ",", "]"))
+            )
+          } else {
+            val d = withEquals
+              .map { case (a1, e1, equal) =>
+                if (equal)
+                  (DarkGray(a1.toString()), DarkGray(e1.toString))
+                else (Red(a1.toString()), Green(e1.toString))
+              }
```

Comment:
Its not important but to get the most of the pattern matching you can reduce the code and make it a little more readable applying the condition into it:
```suggestion
              .map {
                  case (a1, e1, true) => (DarkGray(a1.toString()), DarkGray(e1.toString))
                  case (a1, e1, false) => (Red(a1.toString()), Green(e1.toString))
              }
```

Reply:
@alfonsorr Changed. qq about match expression - is there a case where you would prefer if/else rather than match/case statement? I have a background in Java. That's why it's a new thing for me in the stylistic sense.

Reply:
Well, if it's a case for only a Boolean go for a normal if condition, but in this case that you are deconstructing the tuple you can do more in a single condition. 

I would say that most of only Boolean logic I go for a if condition, but if for example I have multiple booleans that depending on the condition I need a special logic I merge them in a tuple and write a case for each possibly i have.

But always is the one it's more readable for each one hehe